src/Storages/StorageFile.cpp

```diff
@@ -620,8 +623,16 @@ Pipe StorageFile::read(
                 return metadata_snapshot->getColumns();
         };
 
+        /// In case of reading from fd we have to check whether we have already created
```

Comment:
Btw, how do parallel reads from fd work?

Reply:
They don't work I guess, just this branch of code is also executing for files and here we got a data race on moving empty unique ptr (code around schema inference from stdin lools not good, but I had no other variants here)